src/hostapi/alsa/pa_linux_alsa.c

```diff
@@ -2694,7 +2694,7 @@ static PaError PaAlsaStream_DetermineFramesPerBuffer( PaAlsaStream* self, double
     PA_UNLESS( framesPerHostBuffer != 0, paInternalError );
     self->maxFramesPerHostBuffer = framesPerHostBuffer;
 
-    if( !self->playback.canMmap || !accurate )
+    if( (!self->playback.canMmap && self->playback.pcm) || !accurate )
```

Comment:
```suggestion
    if( self->playback.pcm && (!self->playback.canMmap) || !accurate )
```

Check if the struct is valid before checking the `canMmap` field.